controllers/virtualmachine/volumebatch/volumebatch_controller.go

```diff
@@ -355,27 +379,32 @@ func (r *Reconciler) processBatchAttachment(
 			curVolumeAttachSpecMap[volSpec.Name] = volSpec
 		}
 	}
-	var createErrs []error
+
+	var (
+		retErr           error
+		toBeBuiltPvcVols = make([]vmopv1.VirtualMachineVolume, 0)
+	)
+
 	for _, vol := range pvcVolumes {
 		if err := r.handlePVCWithWFFC(ctx, vol, curVolumeAttachSpecMap); err != nil {
-			createErrs = append(createErrs, err)
+			retErr = errOrNoRequeueErr(retErr, err)
+		} else {
+			toBeBuiltPvcVols = append(toBeBuiltPvcVols, vol)
```

Comment:
This is very, very confusing now - we're depending on what a WFFC function returns to even handle volumes that aren't WFFC?

Reply:
Is the naming confusing? If a volume is not WFFC, the function should return nil. Are you recommending that we perform if a volume is WFFC in the main function? I can add a follow-up in the next PR, but would like to get this merge first to unblock others.

Reply:
ack, I am updating https://github.com/vmware-tanzu/vm-operator/pull/1261 to address this.